src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs

```diff
@@ -436,6 +437,33 @@ private async Task StopAsyncCore(bool disposing)
             }
         }
 
+#if NETCOREAPP3_0
+
+        /// <summary>
+        /// Invokes a streaming hub method on the server using the specified method name, return type and arguments.
+        /// </summary>
+        /// <typeparam name="TResult">The return type of the streaming server method.</typeparam>
+        /// <param name="methodName">The name of the server method to invoke.</param>
+        /// <param name="args">The arguments used to invoke the server method.</param>
+        /// <param name="cancellationToken">The token to monitor for cancellation requests. The default value is <see cref="CancellationToken.None" />.</param>
+        /// <returns>
+        /// A <see cref="IAsyncEnumerable{TResult}"/> that represents the stream.
+        /// </returns>
+        public async Task<IAsyncEnumerable<TResult>> StreamAsyncCore<TResult>(string methodName, object[] args, CancellationToken cancellationToken = default)
```

Comment:
Changed this to return a `Task<IAsyncEnumerable>` to preserve the cancellation logic from the `CancelableTypedAsyncEnumerable`

Reply:
To be clear, I'd rather just return the `IAsyncEnumerable` but  otherwise we lose the cancel logic

Reply:
Can you explain this more?

Having StreamAsync return a `Task<IAsyncEnumerable>` *is* more consistent with StreamAsChannelAsync returning `Task<ChannelReader<TResult>>`, but in an ideal world where we had infinite development time neither would return a Task in 3.0.

Is there some reason cancellation cannot work if StreamAsync doesn't return a Task?

Reply:
Before we were doing something similar to 
```csharp
var stream= AsyncEnumerableAdapters.MakeCancelableTypedAsyncEnumerable(enumerable, cancellationToken)
await foreach(var item in stream)
{
    yield return stream
}
```
As I understand it this new AsyncEnumerable that from these returns wont have the  previous cancellationToken that was passed in.

Reply:
You can just move the call to StreamAsChannelCoreAsync into CastIAsyncEnumerable and await it at the beginning. Then StreamAsyncCore doesn't need to be async or Task-returning.